sqldelight-gradle-plugin/src/main/kotlin/com/squareup/sqldelight/gradle/SqlDelightPlugin.kt

```diff
@@ -60,254 +43,49 @@ open class SqlDelightPlugin : Plugin<Project> {
             + "project '${project.path}' but no supported Kotlin plugin was found")
       }
-      val isMultiplatform = project.plugins.hasPlugin("org.jetbrains.kotlin.multiplatform")
-      if (android && !isMultiplatform) {
-        configureAndroid(project, extension)
-      } else {
-        configureKotlin(project, extension, isMultiplatform)
-      }
-    }
-  }
-
-  private fun configureAndroid(
-    project: Project,
-    extension: SqlDelightExtension
-  ) {
-    val variants: DomainObjectSet<out BaseVariant> = when {
-      project.plugins.hasPlugin("com.android.application") -> {
-        project.extensions.getByType(AppExtension::class.java)
-            .applicationVariants
-      }
-      project.plugins.hasPlugin("com.android.library") -> {
-        project.extensions.getByType(LibraryExtension::class.java)
-            .libraryVariants
-      }
-      else -> {
-        throw IllegalStateException("Unknown Android plugin in project '${project.path}'")
-      }
-    }
-    configureAndroid(project, extension, variants)
-  }
-
-  private fun configureKotlin(project: Project, extension: SqlDelightExtension, isMultiplatform: Boolean) {
-    val outputDirectory = File(project.buildDir, "sqldelight")
-
-    val kotlinSrcs = if (isMultiplatform) {
-      val sourceSets = project.extensions.getByType(KotlinMultiplatformExtension::class.java).sourceSets
-      val sourceSet = (sourceSets.getByName("commonMain") as DefaultKotlinSourceSet)
-      project.configurations.getByName(sourceSet.apiConfigurationName).dependencies.add(
-          project.dependencies.create("com.squareup.sqldelight:runtime:$VERSION")
-      )
-      sourceSet.kotlin
-    } else {
-      val sourceSets = project.property("sourceSets") as SourceSetContainer
-      sourceSets.getByName("main").kotlin!!
-    }
-    kotlinSrcs.srcDirs(outputDirectory.toRelativeString(project.projectDir))
-
-    project.afterEvaluate { project ->
-      val packageName = requireNotNull(extension.packageName) { "property packageName must be provided" }
-      val sourceSet = extension.sourceSet ?: project.files("src/main/sqldelight")
-      val className = extension.className ?: DEFAULT_CLASS_NAME
-
-      val ideaDir = File(project.rootDir, ".idea")
-      if (ideaDir.exists()) {
-        val propsDir =
-          File(ideaDir, "sqldelight/${project.projectDir.toRelativeString(project.rootDir)}")
-        propsDir.mkdirs()
-
-        val properties = SqlDelightPropertiesFile(
-            packageName = packageName,
-            sourceSets = listOf(sourceSet.map { it.toRelativeString(project.projectDir) }),
-            outputDirectory = outputDirectory.toRelativeString(project.projectDir),
-            className = className
-        )
-        properties.toFile(File(propsDir, SqlDelightPropertiesFile.NAME))
-      }
 
-      val task = project.tasks.register("generateSqlDelightInterface", SqlDelightTask::class.java) {
```

Comment:
Can we keep this task as an aggregating task which runs all of the db-specific tasks? While not technically part of the public API, I have
```groovy
afterEvaluate {
  compileKotlinJs.dependsOn("generateSqlDelightInterface")
  compileKotlinJvm.dependsOn("generateSqlDelightInterface")
}
```
because my compilation fails without it. When bumping to 1.1.0 now this breaks.

Reply:
Either that or we just need to get the task deps working out-of-the-box. Upon upgrading and commenting these out my sources in commonMain do not see the DB types.